tests/test_builder.py

```diff
@@ -18,18 +18,18 @@ def setUp(self):
             '/a', allow_missing=True), None, config, None)
         self.builder = Builder(None, config)
 
-    def test_get_steps(self):
+    def test_basic(self):
         steps = self.builder.get_steps()
         self.assertEqual(len(steps), 3)
         names = [s.name for s in steps]
         self.assertEqual(names, ['test1', 'test2', None])
 
-    def test_get_steps_with_ref_duplicate(self):
+    def test_ref_duplicate(self):
```

Comment:
Longer test names typically explain more. Something to consider: https://qualitycoding.org/unit-test-names/ . I used another schema in https://github.com/Aalto-LeTech/a-plus/blob/master/shibboleth_login/tests.py and it seems to create relatively readable names.

Reply:
well I wouldn't want there to be tests with different naming conventions, so... should I rename them all? I just thought the class being called 'TestBuilder' was kind of inaccurate and changed it to 'TestGetSteps', and then mentioning 'get_steps' in the function names would've been redundant.

Reply:
You could use `TestBuilderGetSteps` to include class and method, and then not include the method name in test function. Though,  you would like to include the other two parts in it (what is given and what is supposed to happen).

Reply:
You can rename them all, or start using the concept for all future tests.

Reply:
Well I renamed all of them, but I'm not sure how much better the new names are...